dvc/utils/fs.py

```diff
@@ -233,3 +234,21 @@ def walk_files(directory):
     for root, _, files in os.walk(directory):
         for f in files:
             yield os.path.join(root, f)
+
+
+@contextmanager
+def as_atomic(fs, to_info):
+    from dvc.utils import tmp_fname
+
+    tmp_info = to_info.parent / tmp_fname()
+    try:
+        yield tmp_info
+    except BaseException:
+        # Handle stuff like KeyboardInterrupt
+        # as well as other errors that might
+        # arise during file transfer.
+        with suppress(FileNotFoundError):
+            fs.remove(tmp_info)
+        raise
+    else:
+        fs.move(tmp_info, to_info)
```

Comment:
use `finally` here to cover other exceptions?

Reply:
What do you mean by 'cover other exceptions'? `except BaseException:` already covers everything.